tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -599,38 +608,37 @@ def _check_arg_is_anonymous_tuple(self, arg):
 
     return val_key_zipped
 
-  
   @tracing.trace
-  async def _encrypt_client_tensors(self, arg):
+  async def _encrypt_client_tensors(self, arg, pk_a):
 
-    @computations.tf_computation(tf.int32, tf.string)
+    @computations.tf_computation(tf.float32, tf.uint8)
     @tf.function
-    def encrypt_tensor(x, aggregator_key):
-      nonce = tf.random.uniform((), 100, 1000, tf.int32)
-      pk_c = tf.random.uniform((), 100, 1000, tf.int32)
-      sk_c = tf.random.uniform((), 100, 1000, tf.int32)
-      variant = tf.data.Dataset.from_tensor_slices([1]) 
-
-      tf.print("This tensor is encrypted:", x)
-      tf.print("Trusted aggregator public key:", aggregator_key)
-      tf.print("Client public key:", pk_c)
-      tf.print("Client secret key:", sk_c)
-      tf.print("Nonce value: ", nonce)
-      tf.print("I can work with variant", variant._variant_tensor.dtype)
+    def encrypt_tensor(plaintext, aggregator_key):
+      pk_c, sk_c = easy_box.gen_keypair()
+      nonce = easy_box.gen_nonce()
+      ciphertext, mac = easy_box.seal_detached(plaintext, nonce, pk_c, sk_c)
+
+      pk_c.raw.set_shape((32))
```

Comment:
why do you need to do this? guessing it's because the tensors are lacking shape information? if so then we should implement the inference functions on the [ops](https://github.com/tf-encrypted/tf-encrypted/blob/master/primitives/tf_encrypted/primitives/sodium/cc/ops.cc) instead.

Reply:
See https://github.com/capeprivacy/planning/issues/953